From 53e6f0b1623204ef2367859e8c32988d0d4edcad Mon Sep 17 00:00:00 2001
From: Stefan Kost <ensonic@users.sf.net>
Date: Fri, 17 Jul 2009 15:47:10 +0300
Subject: [PATCH] metadata: reset state if there is something to reset

The condition was wrong and as such the element was writing the same tags on subsequent
images if reused.
---
 ext/metadata/gstbasemetadata.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ext/metadata/gstbasemetadata.c b/ext/metadata/gstbasemetadata.c
index 4609013..dd91b2a 100644
--- a/ext/metadata/gstbasemetadata.c
+++ b/ext/metadata/gstbasemetadata.c
@@ -1151,8 +1151,8 @@ gst_base_metadata_calculate_offsets (GstBaseMetadata * base)
     goto done;
   }
 
-  metadata_chunk_array_remove_zero_size (&META_DATA_INJECT_CHUNKS (base->
-          metadata));
+  metadata_chunk_array_remove_zero_size (&META_DATA_INJECT_CHUNKS
+      (base->metadata));
 
   metadata_lazy_update (base->metadata);
 
@@ -1420,7 +1420,7 @@ gst_base_metadata_change_state (GstElement * element, GstStateChange transition)
   switch (transition) {
     case GST_STATE_CHANGE_PAUSED_TO_READY:
       gst_base_metadata_reset_streaming (filter);
-      if (filter->state != MT_STATE_PARSED)
+      if (filter->state == MT_STATE_PARSED)
         gst_base_metadata_reset_parsing (filter);
       break;
     default:
